Upgrade Cake

bdukes opened the pull request:
The build failure for https://github.com/dnnsoftware/Dnn.Platform/pull/4554 is coming from a method signature change between Cake 0.37.0 (which this project currently references) and Cake 1.x (which DNN references).

This PR upgrades the version of Cake referenced by this project to 1.0.0 to avoid these conflicts.

It also adjusts the API to use Cake's path types and to use Cake for logging where appropriate.

Because of the breaking changes in the API surface, this also bumps the version to 2.0.0

bdukes commented:
@donker let me know if there's anything in here you'd like me to adjust.  Thanks!

valadas reviewed (approved):
Looks good to me

donker reviewed (changes requested):
My builds are unfortunately broken with this change. I've discovered some awkward behavior in how Cake deals with paths. If I have a relative path, then new DirectoryPath(path) and then getting FullPath gives back the original relative path, not the expected full path to the target,

bdukes commented:
@donker I've adjusted the places where I'm expecting a full path to use `MakeAbsolute`, and reverted the `name` and `newStart` fields to use `string` instead of `FilePath`/`DirectoryPath`. Let me know if that resolves your issues or if there's anything more that you find needs adjusting.

donker reviewed (approved):
OK, that fixed those issues. We'll go with this.
